ui/helpers/utils/util.js

```diff
@@ -631,3 +633,40 @@ export const getNetworkNameFromProviderType = (providerName) => {
 export const isAbleToExportAccount = (keyringType = '') => {
   return !keyringType.includes('Hardware') && !keyringType.includes('Snap');
 };
+
+/**
+ * Checks if a tokenId in Hex or decimal format already exists in an object.
+ *
+ * @param {*} address - collection address.
+ * @param {*} tokId - tokenId to search for
+ * @param {*} obj - object to look into
+ * @returns {boolean} `false` if tokenId does not already exist.
+ */
+export const tokenIdExist = (address, tokId, obj) => {
+  // check if input tokenId is hexadecimal
+  // If it is convert to decimal and compare with existing tokens
+  // if it is decimal convert to hexadecimal and compare
+  const isHex = isStrictHexString(tokId);
+  let convertedTokenId;
+  if (isHex) {
+    // Convert to decimal
+    convertedTokenId = parseInt(tokId, 16);
+  } else {
+    // Convert to hex
+    const decimalNumber = parseInt(tokId, 10); // 10 is the base for decimal
+    convertedTokenId = `0x${decimalNumber.toString(16)}`;
```

Comment:
Does `decimalToHex` work here?

Reply:
I ll do the same and have it in a separate fct with tests 🙏